libres/tests/analysis/test_assert_update_viable.cpp

```diff
@@ -0,0 +1,143 @@
+#include <filesystem>
+
+#include <ert/analysis/update.hpp>
+#include <ert/enkf/analysis_config.hpp>
+#include <ert/enkf/state_map.hpp>
+#include <ert/enkf/local_updatestep.hpp>
+#include <ert/enkf/local_ministep.hpp>
+#include <ert/analysis/analysis_module.hpp>
+
+#include <catch2/catch.hpp>
+
+namespace analysis {
+bool assert_update_viable(const analysis_config_type *analysis_config,
+                          const state_map_type *source_state_map,
+                          const int ensemble_size,
+                          const local_updatestep_type *updatestep);
+}
+
+TEST_CASE("asserting_update_viable", "[analysis][prototype]") {
+
+    GIVEN("Allocated memory and default value parametrs") {
+        // alloc memory for parameters
+        analysis_config_type *analysis_config = analysis_config_alloc_default();
+        state_map_type *source_state_map = state_map_alloc();
+        local_updatestep_type *updatestep =
+            local_updatestep_alloc("name_not_important");
+        int ensemble_size = 1;
+        analysis_module_type *analysis_module =
+            analysis_module_alloc("not-important");
```

Comment:
I find it a bit surprising that its "not-important" what we call `analysis_module_alloc`.
Should this fail if called with an non-existant analysis-module @pinkwah ?

Reply:
this has been rectify locally should be either `"IES_ENKF" or "STD_ENKF"` but not yet pushed
And no it doesn't fail since it only looks at the `analysis_module_type->get_option` if `local_ministep_type > 1`